### PR TITLE
☸️ K8s Deployment Option

### PR DESCRIPTION
**Pull request by alexchomiak**
☸️  Added K8s deployment configuration files to project (woot)

We are currently using Azure to deploy our project to prod, and due to the free credit that ACM gets from Azure, it makes no sense in moving our current prod setup to k8s in some other cloud provider. Bharat has already done a great job setting up our prod infrastructure. This option is there just for anyone who may want to host the app using k8s!  Plus I learned a lot setting up these configs so it's been a huge benefit to me. 

@bmiddha give me my damn taco

### CODE REVIEW

File: api/src/app.ts [hunk 1]
```diff
@@ -65,7 +65,7 @@ app.use(
         saveUninitialized: true,
         secret: process.env.SESSION_SECRET || 'This is not a secure secret!',
         store: SessionStore,
-        cookie: { secure: process.env.NODE_ENV === 'production' }
+        cookie: { secure: false } // * API Will be served behind reverse proxy, does not need to be secure
```
Comment (kaytwo) [inline api/src/app.ts:68]: The cookie served to the client gets sent "within" the reverse proxy's connection, so it should still be secure. the reverse proxy provides end to end security *in transit*, whereas cookie encryption is more about things outside of either end of the secure pipe, and being secured at rest. this should basically always be true unless you need to do something bonkers like inspect the cookie client side for bugs.

Comment (alexchomiak) [reply api/src/app.ts:68]: No matter what I have tried in NGINX to correctly proxy the cookie, it is not working properly when I set secure to true. Have you ever set up something like this backed by an NGINX proxy and willing to give advice? I'm lost on that problem, I spent a few hours playing around with the NGINX config and was not able to fix it.

Comment (bmiddha) [reply api/src/app.ts:68]: Try using something like
```
server {  proxy_cookie_path / "/; HTTPOnly;   Secure";}
```
References:
http://nginx.org/en/docs/http/ngx_http_proxy_module.html#proxy_cookie_domain
http://nginx.org/en/docs/http/ngx_http_proxy_module.html#proxy_cookie_path

Comment (alexchomiak) [reply api/src/app.ts:68]: @bmiddha I've tried that, didn't work. Trust me I have read tried I can. There are many open issues about this.

Comment (bmiddha) [reply api/src/app.ts:68]: For the current proxy config, it's fine. you can check it in.
I wanna polish this app this summer. We can add a nice edge router (like https://www.envoyproxy.io/ or https://containo.us/traefik/) that will help us resolve this issue and more.

**Review by bmiddha — APPROVED** [review]
Update branch before committing 🌮

**Pull request merged**